### PR TITLE
Test packages and don't rebuild sources

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes https://github.com/aspnet/Coherence-Signed/issues/182

- Run tests without sources
- Remove some old dead code
- Update repo names
- Allow the build to sync repos to particular commits

cc @cesarbs @pranavkm

### CODE REVIEW

File: makefile.shade [hunk 1]
```diff
@@ -120,13 +128,19 @@ var buildTarget = "compile"
...
                 {
-                    foreach (var source in Directory.EnumerateFiles(repoArtifacts, "*.nupkg"))
+                    foreach (var source in Directory.EnumerateFiles(repoBuild, "*.nupkg"))
                     {
-                        File.Copy(source, Path.Combine(universeArtifacts, Path.GetFileName(source)), overwrite: true);
+                        File.Copy(source, Path.Combine(universeBuild, Path.GetFileName(source)), overwrite: true);
+                    }
+                    
+                    var commitFile = Path.Combine(repoArtifacts, "commit");
+                    if (File.Exists(commitFile))
```
Comment (pranavkm) [inline makefile.shade]: Shouldn't this be a failure condition (if the file doesn't exist)?

File: makefile.shade [hunk 2]
```diff
@@ -120,13 +128,19 @@ var buildTarget = "compile"
...
+                    {
+                        commits.TryAdd(repo, File.ReadAllLines(commitFile)[0]);
```
Comment (pranavkm) [inline makefile.shade]: `File.ReadAllText`

File: makefile.shade [hunk 3]
```diff
@@ -151,27 +165,21 @@ var buildTarget = "compile"
                 }
             });
         }
+        
+        var commitsAsString = string.Join(Environment.NewLine, commits.Select(c => c.Key + ":" + c.Value));
```
Comment (pranavkm) [inline makefile.shade]: Can we use `\n` here? Much easier to split it in an environment agnostic way.

File: makefile.shade [hunk 4]
```diff
@@ -679,12 +711,15 @@ functions
 
             return Enumerable.Concat(nonDefaultRepos, repositories);
         }
-
-        bool ShouldSkipCopyingNugetConfig(string repo)
+
```
Comment (JunTaoLuo) [inline makefile.shade]: nit: spacing

File: makefile.shade [hunk 5]
```diff
@@ -151,27 +165,21 @@ var buildTarget = "compile"
...
     }
-
-#smoke-test-mono .pull .fix-project-json .change-default-build-target-to-verify .build-all
-
-#fix-project-json
+    
+#remove-src-folders
     @{
-        repositories = repositories.Except(excludeReposOnMono).ToArray();
+        foreach (var repo in GetRepositoriesToBuild())
```
Comment (pranavkm) [inline makefile.shade]: Deleting sources wouldn't work if projects have `target: project` https://github.com/aspnet/PlatformAbstractions/blob/dev/test/QuickTest/project.json#L8. You might have to additionally address that.

File: makefile.shade [hunk 6]
```diff
@@ -723,6 +758,13 @@ functions
                     return;
                 }
             }
+            
+            if (commitsToSync.ContainsKey(repo))
```
Comment (pranavkm) [inline makefile.shade]: Let's do this as part of the `ci-test`. `#ci-test pull sync-commit ...`

**Comment by pranavkm** [comment]
:watch:

File: makefile.shade (hunk 2 above)
Comment (victorhurdugaci) [inline makefile.shade]: For some reason git adds a new line in the file

File: makefile.shade (hunk 5 above)
Comment (victorhurdugaci) [inline makefile.shade]: Spoke to Andrew and we don't actually need the project reference there. I'll send a PR to change it. Do you know of other repos that use `target: project`

**Comment by victorhurdugaci** [comment]
Updated

File: makefile.shade [hunk 7]
```diff
@@ -23,11 +24,12 @@ functions
         static bool SKIP_NO_CREDENTIALS = Environment.GetEnvironmentVariable("UNIVERSE_SKIP_NO_CREDENTIALS") == "1";
 
         // Doesn't build on Mono since their contracts don't match
-        string[] excludeReposOnMono = new[] { "Helios" };
         string[] repositories = GetRepositoriesToBuild().ToArray();
 
         static bool useHttps = UseHttps(BASE_DIR);
         static string gitHubUriPrefix = useHttps ? "https://github.com/aspnet/" : "git@github.com:aspnet/";
+        
+        static IDictionary<string, string> commitsToSync = GetCommitsToSync();
```
Comment (pranavkm) [inline makefile.shade]: Could we move this into `sync-commits`? Fewer globals

File: makefile.shade [hunk 8]
```diff
@@ -120,14 +141,22 @@ var buildTarget = "compile"
...
                 {
-                    foreach (var source in Directory.EnumerateFiles(repoArtifacts, "*.nupkg"))
+                    foreach (var source in Directory.EnumerateFiles(repoBuild, "*.nupkg"))
                     {
-                        File.Copy(source, Path.Combine(universeArtifacts, Path.GetFileName(source)), overwrite: true);
+                        File.Copy(source, Path.Combine(universeBuild, Path.GetFileName(source)), overwrite: true);
+                    }
+                    
+                    var commitFile = Path.Combine(repoArtifacts, "commit");
+                    if (!File.Exists(commitFile))
```
Comment (pranavkm) [inline makefile.shade]: You could let the `File.ReadAllLines` throw for you.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**